app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,84 @@
+class Api::V1::UsersController < ApplicationController
+  skip_before_action :authenticate_user, only: [:create]
+  before_action :require_admin, only: [:deactivate, :list_all]
+
+  def create
+    result = UsersService.new(user_params: user_params)
+    result.register_user
+    if result.errors.present?
+      render json: { errors: result.errors }, status: :unprocessable_entity
+    else
+      render json:
+        { token: result.authentication_token, message: "User Created Successfully" },
+        status: :created
+    end
+  end
+
+  def update
+    result = UsersService.new(user_params: user_params, current_user: current_user)
+    result.update_user
+    if result.errors.present?
+      render json: { errors: result.errors }, status: :unprocessable_entity
+    else
+      render json: { message: "User Updated Successfully" }, status: :created
+    end
+  end
+
+  def deactivate
+    result = UsersService.new
```

Comment:
`service = UsersService.new`. 
Don't take variable name as `result` because no result is retrieved in line 28. Same for other usages.